src/middlewared/middlewared/plugins/network.py

```diff
@@ -2338,9 +2338,15 @@ async def sync(self):
                     leases = await self.middleware.call('interface.dhclient_leases', interface)
                     reg_routers = re.search(r'option routers (.+);', leases or '')
                     if reg_routers:
-                        # Make sure to get first route only
                         ipv4_gateway = reg_routers.group(1).split(' ')[0]
-                        break
+                        if ipv4_gateway:
```

Comment:
If we do not want to proceed further, is the complete if conditional required starting at L2322 ?